src/openrct2/actions/LargeSceneryRemoveAction.cpp

```diff
@@ -76,20 +76,15 @@ GameActions::Result LargeSceneryRemoveAction::Query() const
         return GameActions::Result(GameActions::Status::Unknown, STR_CANT_REMOVE_THIS, STR_UNKNOWN_OBJECT_TYPE);
     }
 
-    auto rotatedOffsets = CoordsXYZ{
-        CoordsXY{ sceneryEntry->tiles[_tileIndex].x_offset, sceneryEntry->tiles[_tileIndex].y_offset }.Rotate(_loc.direction),
-        sceneryEntry->tiles[_tileIndex].z_offset
-    };
+    auto rotatedOffsets = CoordsXYZ{ CoordsXY{ sceneryEntry->tiles[_tileIndex].offset }.Rotate(_loc.direction),
```

Comment:
I’m sure this can be simplified now that `sceneryEntry->tiles[_tileIndex].offset` is a CoordsXYZ too.

Reply:
you'd think that but sadly not. You need to rotate the x/y and not the z. Afaik this is the usual pattern for that.